src/shared/condition.c

```diff
@@ -353,6 +353,61 @@ static int condition_test_osrelease(Condition *c, char **env) {
         return true;
 }
 
+static int condition_test_dmi(Condition *c, char **env) {
+        int r;
+
+        assert(c);
+        assert(c->type == CONDITION_DMI);
+
+        for (const char *parameter = ASSERT_PTR(c->parameter);;) {
+                _cleanup_free_ char *key = NULL, *condition = NULL, *actual_value = NULL;
+                OrderOperator order;
+                const char *word;
+                bool matches;
+
+                r = extract_first_word(&parameter, &condition, NULL, EXTRACT_UNQUOTE);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to parse parameter: %m");
+                if (r == 0)
+                        break;
+
+                /* parse_order() needs the string to start with the comparators */
+                word = condition;
+                r = extract_first_word(&word, &key, "!<=>", EXTRACT_RETAIN_SEPARATORS);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to parse parameter: %m");
+                /* Ensure env-var-like key names */
+                if (r == 0 || isempty(word))
+                        return log_debug_errno(SYNTHETIC_ERRNO(EINVAL),
+                                        "Failed to parse parameter, key/value format expected: %m");
+
+                /* Do not allow whitespace after the separator */
+                order = parse_order(&word);
+                if (order < 0 || isempty(word) || strchr(WHITESPACE, *word) != NULL)
+                        return log_debug_errno(SYNTHETIC_ERRNO(EINVAL),
+                                        "Failed to parse parameter, key/value format expected: %m");
+
+
+                const char *p = strjoina("/sys/class/dmi/id/", ascii_strlower(key));
+                r = read_one_line_file(p, &actual_value);
+                if (r < 0)
+                        return log_debug_errno(r, "Failed to read %s, ignoring: %m", p);
+
+                /* Might not be comparing versions, so do exact string matching */
+                if (order == ORDER_EQUAL)
+                        matches = streq_ptr(actual_value, word);
+                else if (order == ORDER_UNEQUAL)
+                        matches = !streq_ptr(actual_value, word);
+                else
+                        matches = test_order(strverscmp_improved(actual_value, word), order);
+
+                if (!matches)
+                        return false;
```

Comment:
So, does it really make sense to use ordering and equality operators on DMI fields? The stuff isn't usually something you would want to check order for. Instead, it's probably something to use matches like fnmatch() on, i.e. so that you can for example check for `*Lenovo*` or similar.

Reply:
I think this can be closed